llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -1169,6 +1169,8 @@ instCombineSVEVectorFuseMulAddSub(InstCombiner &IC, IntrinsicInst &II,
     AddendOp = II.getOperand(2);
     Mul = II.getOperand(1);
   }
+  if (match(P, m_ZeroInt()))
+    return IC.replaceInstUsesWith(II, UndefValue::get(II.getType()));
```

Comment:
Is this safe? I can see `instCombineSVEVectorFuseMulAddSub` is called by `instCombineSVEVectorFAdd` whose result for an all inactive predicate is defined.

As with calling `instCombineSVEAllActive` you might need to move this functionality into the relevant places that call this function.